src/Evolution/Particles/MonteCarlo/SwapGrTags.hpp

```diff
@@ -0,0 +1,60 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "Domain/Tags.hpp"
+#include "Evolution/DgSubcell/ActiveGrid.hpp"
+#include "Evolution/DgSubcell/Tags/ActiveGrid.hpp"
+#include "Evolution/DgSubcell/Tags/Inactive.hpp"
+#include "Evolution/DgSubcell/Tags/Mesh.hpp"
+#include "Evolution/Particles/MonteCarlo/System.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+template <size_t Dim>
+class Mesh;
+namespace gsl {
+template <typename T>
+class not_null;
+}  // namespace gsl
+template <typename TagsList>
+class Variables;
+/// \endcond
+
+namespace Particles::MonteCarlo {
+/*!
+ * \brief Swaps the inactive and active GR variables.
+ *
+ * The values on the subcells are at the cell-centers.
+ *
+ * It should be possible to reduce memory usage by deallocating the GR variables
+ * on the DG grid when switching to subcell. However, the opposite case is not
+ * true since the GR variables are needed on the subcells if a neighbor is using
+ * subcell in order to compute the neighbor's fluxes.
+ *
+ * \note The `active_grid` is the grid we are swapping to, which may be the same
+ * as the current grid. On output the `active_gr_vars` will match the grid that
+ * `active_grid` is. This mutator is a no-op if they matched on input.
+ */
+struct SwapGrTags {
```

Comment:
Could you add a test similar to `tests/Unit/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/Test_SwapGrTags.cpp`?